src/braket/pennylane_plugin/translation.py

```diff
@@ -336,8 +339,14 @@ def translate_result_type(
             return DensityMatrix(targets)
         raise NotImplementedError(f"Unsupported return type: {return_type}")
 
-    braket_observable = _translate_observable(observable)
+    if isinstance(observable, qml.Hamiltonian):
+        if return_type is ObservableReturnTypes.Expectation:
+            return tuple(
+                Expectation(_translate_observable(term), term.wires) for term in observable.ops
```

Comment:
Nice.

Reply:
Is [observable grouping](https://pennylane.readthedocs.io/en/stable/code/api/pennylane.grouping.group_observables.html) supported with this new addition? Or the user has to do something beforehand?

Reply:
Observable grouping is only supported for `shots>0` right now. Simultaneous diagonalization is a lot more complex to handle for `shots==0`, and also does not have dramatic benefits as in the `shots>0` case; the reduction in number of tasks offered by grouping doesn't really matter when there is already only a single task.